impl/implqueue/entity_mockmessage.go

```diff
@@ -0,0 +1,14 @@
+package implqueue
+
+type mockRawMessage struct {
+	data []byte
+	idx  int
+}
+
+func (m mockRawMessage) Bytes() []byte {
+	return m.data
+}
+
+func (m mockRawMessage) Metadata() map[string][]byte {
+	return nil
```

Comment:
Why not have metada on the mock implementation?

Reply:
Because it isn't tested... perhaps I should improve the tests